src/compiler/scala/tools/nsc/typechecker/Namers.scala

```diff
@@ -750,6 +750,8 @@ trait Namers extends MethodSynthesis {
       else assignAndEnterFinishedSymbol(tree)
 
       if (isEnumConstant(tree)) {
+        val annots = annotSig(tree.mods.annotations, tree, _ => true)
+        if (annots.nonEmpty) annotate(tree.symbol, annots)
         tree.symbol setInfo ConstantType(Constant(tree.symbol))
```

Comment:
This replaces the lazy type of `tree.symbol`, which would call `MonoTypeCompleter.completeImpl` -> ... -> `valDefSig`.

It would probably be cleaner to move the `if (isEnumConstant(tree)) ...` into `valDefSig`?

Reply:
Tried that, it works for the constant type.

But moving the line below (`linkedClassOfClass addChild tree.symbol`) to `valDefSig` is not OK because evaluation is delayed until forcing the type of the enum value, which can be never, leading to missing exhaustivity warnings.

Reply:
I just refreshed my fading memory of the change from last spring 2.13.11 to populate children of sealed Java classes (that lack a permits clause), which happens in `typedMatch`: all class syms from the file are forced, and the completer will install it as a child if necessary. `typedMatch` could force children of an enum (perhaps recorded in that Permits attachment).

Reply:
I'm trying that out now as a little escape from humdrum life.